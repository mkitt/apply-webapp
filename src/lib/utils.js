// @flow
import { __, addIndex, always, and, complement, either, gt, isEmpty, isNil, map } from 'ramda'
import { withProps } from 'recompose'

// -------------------------------------
// Environments
export const isDev = always(process.env.NODE_ENV === 'development')

// -------------------------------------
// Types
export const isNotNil = complement(isNil)
export const isNotEmpty = complement(isEmpty)
export const isEmptyOrNil = either(isNil, isEmpty)
export const isNotEmptyAndNotNil = and(isNotNil, isNotEmpty)

// -------------------------------------
// Relation
export const gt0 = gt(__, 0)

// -------------------------------------
// Lists
export const nool = always([])()
export const mapIndexed = addIndex(map)

// -------------------------------------
// Objects
export const noob = always({})()

// -------------------------------------
// Functions
export const noop = () => {}

export const log = withProps(props => (
  // eslint-disable-next-line no-console
  console.log(props.displayName || 'üëç', props) && props
))

export const warn = withProps(props => (
  // eslint-disable-next-line no-console
  console.warn(props.displayName || '‚ö†Ô∏è', props) && props
))
