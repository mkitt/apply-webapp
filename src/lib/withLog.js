import { withProps } from 'recompose'

export default withProps(props => (
  // eslint-disable-next-line
  console.log(props.logName || props.displayName || '👍', props) && props
))
