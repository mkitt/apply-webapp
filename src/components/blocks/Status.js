// @flow
import React, { type Node } from 'react'
import glamorous from 'glamorous'
import { alignItems, space } from 'styled-system'
import { isNotNil } from '../../lib/utils'
import Button from '../elements/Button'
import { RocketIcon } from '../elements/Icons'
import { COLORS } from '../../constants/theme'

type Props = {
  id: string,
  children: Node,
  onClick?: ?() => void,
}

const Status = ({ id, children, onClick, ...props }: Props) => (
  <Dl {...props}>
    <Dt>Application Status</Dt>
    <Dd>
      { isNotNil(onClick) ?
        <Button box centered primary onClick={onClick} data-id={id}>
          {children}
        </Button> :
        <Text>
          <RocketIcon marginRight={8} />
          {children}
        </Text>
      }
    </Dd>
  </Dl>
)

Status.defaultProps = {
  onClick: null,
}

export default Status

// -------------------------------------

const Dl = glamorous.dl(
  {
    position: 'relative',
    zIndex: 2,
    display: 'flex',
    flexDirection: 'column',
    marginTop: 8,
    marginBottom: 8,
  },
  alignItems,
  space,
)

const Dt = glamorous.dt({
  fontSize: 14,
  fontWeight: 600,
  textTransform: 'uppercase',
  color: COLORS.GREY_9,
})

const Dd = glamorous.dd({
  marginTop: 8,
  marginLeft: 0,
})

const Text = glamorous.span({
  display: 'flex',
  alignItems: 'center',
  marginTop: 8,
  fontSize: 14,
  fontWeight: 400,
  textTransform: 'uppercase',
  color: COLORS.GREY_4,
})
