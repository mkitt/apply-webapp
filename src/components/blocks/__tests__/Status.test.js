import React from 'react'
import renderer from 'react-test-renderer'
import { noop } from '../../../lib/utils'
import Status from '../Status'

describe('<Status />', () => {
  it('shows the correct snapshot tree for the <Status> block (with click)', () => {
    const component = (
      <Status onClick={noop}>
        Status children with click
      </Status>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <Status> block (without click)', () => {
    const component = (
      <Status>
        Status children no click
      </Status>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
