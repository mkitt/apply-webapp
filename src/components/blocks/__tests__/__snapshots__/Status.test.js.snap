// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> shows the correct snapshot tree for the <Status> block (with click) 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 8px;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-1,
[data-css-1] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  color: #999;
}

.css-2,
[data-css-2] {
  margin-top: 8px;
  margin-left: 0;
}

.css-3,
[data-css-3] {
  text-align: center;
  min-width: 240px;
  height: 48px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 14px;
  text-transform: uppercase;
  color: #fff;
  background-color: #4d4d4d;
}

<dl
  className="css-0"
>
  <dt
    className="css-1"
  >
    Application Status
  </dt>
  <dd
    className="css-2"
  >
    <button
      className="css-3"
      data-id={undefined}
      onClick={[Function]}
      role="button"
    >
      Status children with click
    </button>
  </dd>
</dl>
`;

exports[`<Status /> shows the correct snapshot tree for the <Status> block (without click) 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  margin-top: 8px;
  margin-bottom: 8px;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-1,
[data-css-1] {
  font-size: 14px;
  font-weight: 600;
  text-transform: uppercase;
  color: #999;
}

.css-2,
[data-css-2] {
  margin-top: 8px;
  margin-left: 0;
}

.css-3,
[data-css-3] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  margin-top: 8px;
  font-size: 14px;
  font-weight: 400;
  text-transform: uppercase;
  color: #4d4d4d;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.css-4,
[data-css-4] {
  margin-right: 8px;
}

<dl
  className="css-0"
>
  <dt
    className="css-1"
  >
    Application Status
  </dt>
  <dd
    className="css-2"
  >
    <span
      className="css-3"
    >
      <svg
        className="css-4"
        height={32}
        viewBox="0 0 24 24"
        width={32}
      >
        <g
          fill="currentColor"
        >
          <path
            d="M24,0 C24,0 23.4836048,9.80252996 16.4820958,15.2719707 L16.4128111,19.0346251 L11.4820194,24 L11.4746834,20.2882431 L9.63105705,18.6292449 C8.29671435,22.7691304 1.26439406,22.7691304 1.26439406,22.7691304 C1.26439406,22.7691304 1.26439406,16.3396736 5.58319762,14.4450082 L3.94220876,12.6079505 L-2.84217094e-14,12.6079505 L5.07440302,7.66843342 L8.7285621,7.58627726 C14.4472436,0.671411685 24,0 24,0 Z M16.425,10.1644246 C17.8333261,10.1644246 18.975,9.01830529 18.975,7.60449544 C18.975,6.19068562 17.8333261,5.0445663 16.425,5.0445663 C15.0166739,5.0445663 13.875,6.19068562 13.875,7.60449544 C13.875,9.01830529 15.0166739,10.1644246 16.425,10.1644246 Z"
          />
        </g>
      </svg>
      Status children no click
    </span>
  </dd>
</dl>
`;
